natalie/README.md

```diff
@@ -0,0 +1,227 @@
+
+**a)**
+Write an expression for the mean (i.e. average) of x, y, and z.
+
+`var mean = (x+y+z)/3`
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward the mean of the three.
+That is, reset the value of each variable to something new based on its previous value.
+
+```
+x = x - ( x - mean )/2
+y = y - ( y - mean )/2
+z = z - ( z - mean )/2
+```
+
+**3)** (_20%_)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+- _l_ : horizontal position of left edge (relative to some origin);
+- _r_ : horizontal position of right edge;
+- _t_ : vertical position of top edge;
+- _b_ : vertical position of bottom edge.
+
+**a)**
+Write an expression for the rectangle's area.
+
+`( r - l ) * ( t - b )`
+
+or
+
+```
+var width = r - l
+var height = t - b
+var area = width * height
+```
+
+**b)**
+Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+
+```
+if (height > width) {
+  true;
+} else {
+  false;
+}
+```
+
+**c)**
+Write an expression for the circumference of the biggest circle which can fit inside the rectangle.  (Hint: you'll need logic similar to that in **b**.)
+
+```
+if ( width <= height ) {
+  circumference = width * Math.PI;
+} else {
+  circumference = height * Math.PI;
+}
+```
+
+**d)**
+Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+
+Knowing that the area of a circle is (radius^2) * Pi:
+
+```
+var radiusSquared = Math.pow(width/2, 2) + Math.pow(height/2, 2)
+var smallestCircleArea = radiusSquared * Math.PI
+```
+
+**e)**
+Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+Define four new variables describing the centermost small rectangle.
+(_Hint: one of the many solutions is very similar to the solution of **2b** above._)
+
+```
+var height = t - b
+var width = r - l
+var new_t = t - height/3
+var new_b = b + height/3
+var new_r = r - width/3
+var new_l = l + width/3
+```
+
+**4)** (_25%_)
+
+Imagine that the squares of an ordinary checkerboard are numbered in two different ways:
+
+* Each square has integer coordinates _(R,C)_ describing its row and column.  Both values should be in the range 0..7, so that the upper-left square is at (0,0) and the bottom-right is at (7,7).
+
+* Each square has a unique integer number N from 0 to 63.  These numbers run sequentially left-to-right one row at a time, top to bottom.  Therefore the upper-left square has N===0 and the bottom-right has N===63.
+
+**a)**  Given a particular R and C, find the corresponding N.  That is, write an expression for variable N containing variables R and C.
+
+`N = ( R * 7 ) + ( R + C )`
+
+**b)**  Given N, find R.  Write an expression for R which contains N.
+
+`R = ( N - 8 ) % 8`
+
```

Comment:
Close again! The function math.floor is really helpful here: (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/floor). Essentially what it does is to round the number down to the nearest whole number. Therefore, if we divided N by 8, and then used math.floor, it would give us the proper row. The full solution would look like this: 

r = Math.floor(n/8);